tools/js-tools/eslintrc/base.mjs

```diff
@@ -325,6 +325,18 @@ export default function makeBaseConfig( configurl, opts = {} ) {
 					'error',
 					{ allowInterfaces: 'with-single-extends' },
 				],
+				// Disable param name check for destructured params.
+				'jsdoc/check-param-names': [ 'error', { checkDestructured: false } ],
+				// TS should be enough to catch undefined types.
+				'jsdoc/no-undefined-types': 'warn',
+				// Disable param check except when destructured.
+				'jsdoc/require-param': [ 'error', { checkDestructured: false } ],
```

Comment:
Why are you disabling this? jsdoc's `recommended-typescript` config doesn't do this.

What this seems to disable is it warning about the missing `@param opts.foo` in something like this:
```ts
/**
 * @param opts - Options
 * @param opts.bar - Bar
 * @returns xxx
 */
function xxx( { foo: string, bar: number } ): null {
    return null;
}
```
Which doesn't seem to have anything to do with types.

Reply:
It disables it only when the props are restructured, e.g. in component props; otherwise, it maintains the default behavior.

Reply:
Otherwise this will have that error

```ts
/**
 * Desc.
 *
 * @param opts - Options
 *
 * @return xxx
 */
function xxx( { foo: string } ): null {
	return null;
}
```

<img width="828" alt="image" src="https://github.com/user-attachments/assets/6b6a1de9-8fe6-4468-9cc0-e2c4e37f7f51" />

Reply:
Because I can and would prefer to document it like this

```ts
type FooProps = {
	/**
	 * Description for foo.
	 */
	foo: string;
};

/**
 * Desc.
 *
 * @param props - Component props.
 *
 * @return Rendered Foo
 */
function Foo( { foo: string }: FooProps ): null {
	return null;
}
```

Reply:
Alright, I have removed that for now.